app/models/role.rb

```diff
@@ -251,6 +252,15 @@ def self.search_by_permission(key, operator, value)
     {:conditions => role_condition}
   end
 
+  def self.search_by_locked(key, operator, value)
+    if value == 'true'
+      role_condition = "origin IS NOT NULL AND builtin <> #{BUILTIN_DEFAULT_ROLE}"
+    else
+      role_condition = "origin IS NULL OR builtin = #{BUILTIN_DEFAULT_ROLE}"
```

Comment:
While this is correct, I'd prefer picking one and just negating it if we need the other instead of having two distinct queries

Reply:
Also in general we should avoid directly interpolating values into strings being passed on to the database to prevent sql injection attacks, but considering we are passing in a constant under our control, we can let it slide here.

Reply:
Thanks, added a `NOT` before the false one